service/backend/common/parser.go

```diff
@@ -0,0 +1,22 @@
+package common
+
+import (
+	"github.com/ava-labs/avalanchego/codec"
+	"github.com/ava-labs/avalanchego/utils/wrappers"
+	"github.com/ava-labs/avalanchego/vms/platformvm/txs"
+)
+
+// initializes tx to have tx identifier generated
+func InitializeTx(version uint16, c codec.Manager, tx *txs.Tx) error {
```

Comment:
nit: seems only used in pchain package. Why is it in common?

Reply:
Initially planned to use for both p and c chain, but it didn't work out that way. Currently used by pchain block and indexer parser. Will copy it to the respective files considering it is few lines and we can use sane default values for version and codec.